### PR TITLE
[OptApp] Adding magnitude reduction convergence criterion

### PR DESCRIPTION
**Pull request by talhah-ansari**
**📝 Description**
Adding magnitude reduction convergence criterion

Please mark the PR with appropriate tags: 

**🆕 Changelog**

Added 'magnitude_reduction' class

### CODE REVIEW

File: applications/OptimizationApplication/python_scripts/utilities/opt_convergence.py [hunk 1]
```diff
@@ -181,4 +183,49 @@ def GetInfo(self) -> dict:
...
+        iter = self.__optimization_problem.GetStep()
+        self.conv = iter >= self.__max_iter
+
+        algorithm_buffered_data = ComponentDataView("algorithm", self.__optimization_problem).GetBufferedData()
+        if not algorithm_buffered_data.HasValue("std_obj_value"):
+            raise RuntimeError(f"Algorithm data does not contain computed \"std_obj_value\".\nData:\n{algorithm_buffered_data}" )
+        self.value = algorithm_buffered_data["std_obj_value"]
+
+        if iter == 0:
+            #self.__initial_value = self.value
```
Comment (sunethwarna) [inline applications/OptimizationApplication/python_scripts/utilities/opt_convergence.py]:
```suggestion
```

**Review by sunethwarna — COMMENTED** [review]
Thanks @talhah-ansari . I'll leave the approval to @Igarizza

File: applications/OptimizationApplication/python_scripts/utilities/opt_convergence.py (hunk 1 above)
Comment (talhah-ansari) [reply applications/OptimizationApplication/python_scripts/utilities/opt_convergence.py]: Done

**Review by Igarizza — COMMENTED** [review]
@talhah-ansari thank you for your work. Please, add the unit test to the PR. You can find in algorithms_test test_convergence.py

**Pull request closed without merging**